src/main/resources/reference.conf

```diff
@@ -123,6 +177,230 @@ resourceTypes = {
     authDomainConstrainable = true
     reuseIds = false
   }
+  controlled-user-shared-workspace-resource = {
+    actionPatterns = {
+      delete = {
+        description = "delete this resource"
+      }
+      read_policies = {
+        description = "view all policies and policy details for this resource"
+      }
+      set_parent = {
+        description = "set the parent workspace of this resource"
+      }
+      "share_policy::owner" = {
+        description = "change the membership of the owner policy for this resource"
+      }
+      "share_policy::editor" = {
+        description = "change the membership of the editor policy for this resource"
+      }
+      "share_policy::writer" = {
+        description = "change the membership of the writer policy for this resource"
+      }
+      "share_policy::reader" = {
+        description = "change the membership of the reader policy for this resource"
+      }
+      "read" = {
+        description = "perform reader actions on the resource"
+      }
+      "write" = {
+        description = "perform writer actions on the resource"
+      }
+      "edit" = {
+        description = "perform editor actions on the resource"
+      }
+      "own" = {
+        description = "perform owner actions on the resource"
+      }
+    }
+    ownerRoleName = "owner"
+    roles = {
+      owner = {
+        roleActions = ["delete", "read_policies", "share_policy::owner", "share_policy::editor", "share_policy::writer", "share_policy::reader", "own", "edit", "set_parent"]
+      }
+      editor = {
+        roleActions = ["delete", "edit"]
+      }
+      writer = {
+        roleActions = ["write"]
```

Comment:
I wonder if a better way to organize this is to have the writer have "read" and "write" actions. Then have the writer role in the workspace just map to the writer role in the resource.

Reply:
If we want all writers to also be readers, then yeah that's better. 
Controlled resources don't have the tiered access like workspaces do (e.g. editors explicitly don't have reader/writer access) so I was hesitant to add that here, but it's fine if that's how we want to structure it

Reply:
Good point. Let's leave it just as it is.

Reply:
When in doubt, our philosophy in Sam has been to have roles contain the minimal set of information, because we can always grant multiple roles. I think leaving as is seems like the right call.